pkg/entdbadapter/hooks.go

```diff
@@ -0,0 +1,228 @@
+package entdbadapter
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/fastschema/fastschema/db"
+	"github.com/fastschema/fastschema/schema"
+)
+
+func runPreDBQueryHooks(ctx context.Context, client db.Client, option *db.QueryOption) error {
+	if client == nil {
+		return nil
+	}
+
+	hooks := client.Hooks()
+	if len(hooks.PreDBQuery) > 0 {
+		for _, hook := range hooks.PreDBQuery {
+			if err := hook(ctx, option); err != nil {
+				return fmt.Errorf("pre query hook: %w", err)
+			}
+		}
+	}
```

Comment:
Why do you add  `if len(hooks.PreDBQuery) > 0` to this case? I think you can do like this to avoid nested code:
```suggestion
   for _, hook := range client.Hooks().PreDBQuery {
        if err := hook(ctx, option); err != nil {
            return fmt.Errorf("pre query hook: %w", err)
        }
    }
```

Reply:
@conglt-2778 the client Hooks may be nil, we should check before using it